### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Arjunmdas3211**
new branch develop with new features

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -6,4 +6,7 @@ body{
 a {
     color:black;
     text-decoration:none;
+}
+*{
```
Comment (amouxkin) [inline src/App.css:10]: Use normalize.css or something smilar

File: src/App.js [hunk 2]
```diff
@@ -6,22 +6,54 @@ import Header from './components/header/header.component';
 import SignInAndSignUpPage from './pages/Sing-In-Sign-Up/Sing-In-Sign-Up.component';
 
 
-
+import {auth, createUserProfileDocument } from './firebase/firebase.utils'; 
 
 import './App.css';
-function App() {
-  return (
+class  App extends React.Component {
```
Comment (amouxkin) [inline src/App.js]: Leave a space after import.

File: src/App.js [hunk 3]
```diff
@@ -6,22 +6,54 @@ import Header from './components/header/header.component';
...
+  constructor(props){
+    super(props);
+
+    this.state = {
+      currentUser:null
```
Comment (amouxkin) [inline src/App.js]: space after `:`.

No need to set null values, only useful when using typescript.

File: src/App.js [hunk 4]
```diff
@@ -6,22 +6,54 @@ import Header from './components/header/header.component';
...
+    };
+  }
+  unsubscribeFromAuth = null;
```
Comment (amouxkin) [inline src/App.js:17]: No need to set to null. Just declaration is fine.

File: src/App.js [hunk 5]
```diff
@@ -6,22 +6,54 @@ import Header from './components/header/header.component';
...
+
+  componentDidMount() {
+    this.unsubscribeFromAuth = auth.onAuthStateChanged(async userAuth => {
```
Comment (amouxkin) [inline src/App.js:21]: Why do it like this ?
Why not implement it in declaration ?

File: src/components/collection-item/collection-item.component.jsx [hunk 6]
```diff
@@ -2,7 +2,8 @@ import React from 'react';
 
 import './collection-item.styles.scss';
 
-const CollectionItem = ({id,name,price,imageUrl}) => (
+// eslint-disable-next-line react/prop-types
```
Comment (amouxkin) [inline src/components/collection-item/collection-item.component.jsx]: Don't disable eslinting.

File: src/components/form-input/form-input.component.jsx [hunk 7]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 
 import './form-input.styles.scss';
 
-const  FormInput = ({handlechange , label , ...otherProps}) =>(
+const  FormInput = ({handlechange, label, ...otherProps}) =>(
```
Comment (amouxkin) [inline src/components/form-input/form-input.component.jsx:5]: no need to specify as `otherPros`, just `props` is fine. It's understood.

File: src/components/custom-button/custom-button.component.jsx [hunk 8]
```diff
@@ -2,9 +2,10 @@ import React from'react';
 
 import './custom-button.styles.scss';
 
-const CustomButton = ({children , ...otherProps}) => (
+// eslint-disable-next-line react/prop-types
+const CustomButton = ({children, isGoogleSignIn, ...otherProps}) => (
 
-    <button className='custom-button' {...otherProps}>
+    <button className={`custom-button ${isGoogleSignIn ? 'google-sign-in': ''}`} {...otherProps}>
```
Comment (amouxkin) [inline src/components/custom-button/custom-button.component.jsx:8]: try this without `children` restructuring and check if that works.

File: src/components/sign-in/sign-in.component.jsx [hunk 9]
```diff
@@ -15,10 +15,18 @@ class SignIn extends React.Component{
...
 
 }
-handleSubmit=  event =>{
+handleSubmit= async event =>{
     event.preventDefault();
-
-    this.state({email:'', password:''});
+    const {email,password} =this.state;
+
+    try{
```
Comment (amouxkin) [inline src/components/sign-in/sign-in.component.jsx:21]: it's async, so use `.catch` method, more cleaner.

File: src/components/sign-in/sign-in.component.jsx [hunk 10]
```diff
@@ -15,10 +15,18 @@ class SignIn extends React.Component{
...
+        await auth.signInWithEmailAndPassword(email, password);
+        this.setState({email:'', password:''});
```
Comment (amouxkin) [inline src/components/sign-in/sign-in.component.jsx:23]: why set empty strings here ?

File: src/components/sign-in/sign-in.component.jsx [hunk 11]
```diff
@@ -32,12 +40,17 @@ render(){
...
             
-            <FormInput type="password" name='email' label='Password' value={this.state.password} handleChange={this.handleChange} required/>
+            <FormInput type="password" name="password" label="Password" value={this.state.password} handlechange={this.handleChange} required/>
             
-
-            <CustomButton type='submit'>Sign In</CustomButton>
+            <div className="buttons">
+            <CustomButton type="submit">Sign In</CustomButton>
+            <CustomButton onClick ={signInWithGoogle} isGoogleSignIn>
+              {''} Sign In With Google{''}
```
Comment (amouxkin) [inline src/components/sign-in/sign-in.component.jsx]: Why `{''}`?

File: src/components/sign-up/sign-up.component.jsx [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+         alert("password don't match")
+         return; 
+       }
+       try{
+           const {user} = await auth.createUserWithEmailAndPassword(email, password);
+           
+           await createUserProfileDocument(user, {displayName});
+
+           this.setState({
+            displayName: '',
```
Comment (amouxkin) [inline src/components/sign-up/sign-up.component.jsx:34]: indent missing.

File: src/components/sign-up/sign-up.component.jsx [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+           password:'',
+           confirmPassword: ''
+       }
+   }
+
+   handleSubmit = async event => {
+       event.preventDefault();
+
+       const {displayName, email, password, confirmPassword}= this.state;
+       if(password!==confirmPassword){
```
Comment (amouxkin) [inline src/components/sign-up/sign-up.component.jsx:24]: fix spacing.

**Review by amouxkin — CHANGES_REQUESTED** [review]

File: src/components/sign-in/sign-in.component.jsx (hunk 10 above)
Comment (Arjunmdas3211) [reply src/components/sign-in/sign-in.component.jsx:23]: just test so after login just clear the feild . After completion convert it to link to main page.

**Comment by amouxkin** [comment]
Close and merge this branch.

**Pull request merged**